ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -1292,95 +1334,75 @@ void instr_execute_DIELECTRIC(in instr_t instr, in uvec3 regs, in mat2 dUV, in b
 }
 void instr_execute_ROUGHDIELECTRIC(in instr_t instr, in uvec3 regs, in mat2 dUV, in bsdf_data_t data)
 {
-	uint ndf = instr_getNDF(instr);
 	float eta = uintBitsToFloat(data.data[0].x);
-	vec3 scale = decodeRGB19E7(uvec2(data.data[1].w, data.data[2].x));
-	float au = textureOrF32(data.data[0].yz, instr_getAlphaUTexPresence(instr), dUV)*scale.x;
-	float av;
-	if (ndf == NDF_AS)
-		av = textureOrF32(uvec2(data.data[0].w, data.data[1].x), instr_getAlphaVTexPresence(instr), dUV)*scale.y;
 	registers[REG_DST(regs)] = reg_t(1.0, 0.0, 0.0);
 }
 void instr_execute_CONDUCTOR(in instr_t instr, in uvec3 regs, in mat2 dUV, in bsdf_data_t data)
 {
-	vec3 scale = decodeRGB19E7(data.data[2].zw);
-	vec3 eta = decodeRGB19E7(data.data[1].xy);
-	vec3 etak = decodeRGB19E7(data.data[1].zw);
+	vec3 eta = decodeRGB19E7(data.data[2].xy);
+	vec3 etak = decodeRGB19E7(data.data[2].zw);
 	registers[REG_DST(regs)] = reg_t(1.0, 0.0, 0.0);
 }
-void instr_execute_ROUGHCONDUCTOR(in instr_t instr, in uvec3 regs, in mat2 dUV, in vec2 a, in float D, in bsdf_data_t data)
+void instr_execute_ROUGHCONDUCTOR(in instr_t instr, in uvec3 regs, in mat2 dUV, in float DG, in bsdf_data_t data)
 {
 	if (currBSDFParams.isotropic.NdotL>FLT_MIN)
 	{
 		mat2x3 eta2;
-		eta2[0] = decodeRGB19E7(data.data[1].xy); eta2[0]*=eta2[0];
-		eta2[1] = decodeRGB19E7(data.data[1].zw); eta2[1]*=eta2[1];
-		float a2 = a.x*a.y;
+		eta2[0] = decodeRGB19E7(data.data[2].xy); eta2[0]*=eta2[0];
+		eta2[1] = decodeRGB19E7(data.data[2].zw); eta2[1]*=eta2[1];
 		vec3 fr = irr_glsl_fresnel_conductor(eta2[0],eta2[1],currBSDFParams.isotropic.VdotH);
-		float g = irr_glsl_beckmann_smith_height_correlated(currBSDFParams.isotropic.interaction.NdotV_squared, currBSDFParams.isotropic.NdotL_squared, a2);
-		registers[REG_DST(regs)] = g*D*fr / (4.0 * currBSDFParams.isotropic.interaction.NdotV);
+		registers[REG_DST(regs)] = DG*fr;
 	}
 	else registers[REG_DST(regs)] = reg_t(0.0);
 }
-void instr_execute_PLASTIC(in instr_t instr, in uvec3 regs, in mat2 dUV, in bsdf_data_t data)
+void instr_execute_PLASTIC(in instr_t instr, in uvec3 regs, in mat2 dUV, in vec3 scale, in bsdf_data_t data)
 {
-	float eta = uintBitsToFloat(data.data[0].x);
-	vec3 scale = decodeRGB19E7(uvec2(data.data[1].w, data.data[2].x));
-	vec3 refl = textureOrRGB19E7(data.data[1].yz, instr_getPlasticReflTexPresence(instr), dUV)*scale.z;
+	float eta = uintBitsToFloat(data.data[2].x);
+	vec3 refl = textureOrRGB19E7(data.data[1].zw, instr_getPlasticReflTexPresence(instr), dUV)*scale.y;
 	registers[REG_DST(regs)] = refl;
```

Comment:
this now needs to be modulated by `irr_glsl_diffuseFresnelCorrectionFactor(eta,eta2) * (vec3(1.0)-irr_glsl_fresnel_dielectric(eta, currBSDFParams.isotropic.interaction.NdotV)) * (vec3(1.0)-irr_glsl_fresnel_dielectric(eta, currBSDFParams.isotropic.NdotL))` just like `ROUGHPLASTIC`

Ideally you'd move this (and the reflectance texture fetch) outside of both or combine the two functions to reduce divergence

Reply:
let me know if tackled in new implementation.

Reply:
?

Reply:
yes, irrelevant now pretty much